feat(common): add a geocoder search to the search-combobox

clintonlunn opened the pull request.

Mackenzie-Cope commented:
Talked with @clintonlunn, we'd like to implement:

- zooming after a selection
- zoom level based on feature type: polygon (extent of the polygon), line (extent of the line), point (neighborhood-city level), _pull any following searches back to the standard zoom level every time_
- Look into keeping the search window open when a user hits enter (so that they know they much select an address)

clintonlunn commented:
@Mackenzie-Cope I fixed the points from yesterday's meeting 
![image](https://github.com/user-attachments/assets/b07d70e9-718c-4c8c-b676-2b73668ece27)


the various bugs comment was related to keyboard navigation not working as expected. i have changed things to have the dropdown highlighting work with keyboard navigation and selection

Mackenzie-Cope commented:
When chatting with Clinton, we found these last things to address: 

1. The search bar displays a single selected feature but searches that are not a specific feature (many) does not display in the top bar.
2. Still seeing the bug where the zoom and pan takes the user to the Mediterranean. Logging as a bug, needs more work.
3. API search is not zooming to the default level (CCUS)
4. Mobile - don't let the search bar push the theme switcher off the screen

clintonlunn commented:
zooming on ccus broken, move shaking to words and add tooltip to help user

Mackenzie-Cope reviewed (approved):
Looks fantastic, nice work. I like the prompts to the user, it's really slick. I have no more comments.
